tests/tests_lvm_errors.yml

```diff
@@ -149,3 +149,89 @@
                       not blivet_output.changed }}"
             msg: "Unexpected behavior w/ LVM volume defined outside of any pool"
       ignore_errors: yes
+
+    - name: Test for correct handling of safe_mode
+      block:
+        - name: Create a pool
+          include_role:
+            name: storage
+          vars:
+            storage_pools:
+              - name: testpool1
+                type: lvm
+                disks: "{{ unused_disks }}"
+                volumes:
+                  - name: testvol1
+                    fs_type: 'ext4'
+                    size: '1g'
+
+        - name: Try to replace file system in safe mode
+          include_role:
+            name: storage
+          vars:
+            storage_safe_mode: true
+            storage_pools:
+              - name: testpool1
+                type: lvm
+                disks: "{{ unused_disks }}"
+                volumes:
+                  - name: testvol1
+                    fs_type: 'ext3'
+                    size: '1g'
+
+        - name: Verify the output
+          assert:
+            that: "{{ blivet_output.failed and
+                      blivet_output.msg|regex_search('cannot remove existing formatting on volume.*in safe mode') and
+                      not blivet_output.changed }}"
+            msg: "Unexpected behavior w/ existing data on specified disks"
+
+        - name: Try to resize in safe mode
+          include_role:
+            name: storage
+          vars:
+            storage_safe_mode: true
+            storage_pools:
+              - name: testpool1
+                type: lvm
+                disks: "{{ unused_disks }}"
+                volumes:
+                  - name: testvol1
+                    fs_type: 'ext4'
+                    size: '2g'
```

Comment:
Why should safe mode prevent resizing of a volume? Is it a destructive operation?

Reply:
If it goes wrong it can be ;)

Reply:
Isn't it the case for every storage operation? Is the risk particularly high for resizing?
For me resizing is a common operation and if it forces me to turn safe mode off, it just increases the likelihood that I will overwrite some valid data by accident. (Yes, if resizing is especially risky for some reason, it should be made conditional on safe mode being off, but this sounds like a deeper bug in the storage stack in the first place.)

Reply:
The only difference is the possibility of losing pre-existing data you wanted to keep. I am open to dropping resize from this patch if the consensus is that it's not appropriate. I was just trying to make it really safe for the people who are into safety. (I'm not generally one of them.)

Reply:
Sorry, I still don't get it. Why should resizing result in losing pre-existing data one wants to keep? When I use resize2fs or xfs_growfs, I don't consider them as unsafe operations.

Reply:
The difference is this: if you are removing a device, you don't care about the data, so if the operation fails it does not matter. If you are creating a device there is no data to worry about losing if the operation goes awry. However, if you are resizing and things go sideways there is data involved that you presumably wanted to keep -- otherwise you would probably not be resizing.

Reply:
I would say those are two very different meanings of "safe". When the role removes existing filesystems and physical volumes, it is sensitive to user errors. storage_safe_mode then protects users from their mistakes - when they accidentally asked for a filesystem to be created on a disk which contains valuable data instead of on an empty one.
In case of resizing, the only thing this setting protects from is a bug in the storage stack. Then, if we do not trust the storage stack enough, the functionality to resize filesystems should simply not be there, and if we trust it, it should not be governed by storage_safe_mode.
Given this difference, I really prefer storage_safe_mode to guard against overwrites of existing data only.

Reply:
I.e. require `storage_safe_mode: false` mostly in situations where various management utilities already require some kind of --force flag.

Reply:
That sounds fine to me: `storage_safe_mode` is about preventing users from accidentally removing their existing data. I will update the code and the descriptions accordingly.

Reply:
@dwlehman thanks, and what is your opinion on enabling it by default? Otherwise there is a risk that the user will learn about in only when it is too late.